api-2.0/src/main/java/org/openmrs/module/attachments/obs/ComplexViewHelper2_0.java

```diff
@@ -29,25 +29,21 @@ public String getView(Obs obs, String view) {
 		// https://issues.openmrs.org/browse/ATT-34
 		
 		ComplexObsHandler complexObsHandler = context.getObsService().getHandler(obs);
-		String [] supportedViews = complexObsHandler.getSupportedViews();
 		
-		for(int x=0;x<supportedViews.length;x++){
-			if(supportedViews[x].equalsIgnoreCase(view)){
-				return view;
-			}
-		} 
-			log.warn("Supported view for this Obs not found.Obs will now use the default view");
-			return ComplexObsHandler.RAW_VIEW;
-		
-		/*
-		complexObsHandler.getSupportedViews();
 		if (complexObsHandler.supportsView(view)) {
 			return view;
 		} else {
 			log.warn("Supported view for this Obs not found.Obs will now use the default view");
```

Comment:
Change this to something that helps better the troubleshooter:
```java
"'" + view + "' is not a supported view for obs (" + obs.getUuid() + "). Reverting to the default view: '" + ComplexObsHandler.RAW_VIEW + "'".
```

Reply:
Thanks @mks-d I am going to write the test...However it would be super helpful if you could advice on why the current test is failing.

Reply:
@reagan-meant this is even better/clearer:
```java
"'" + view + "' is not a supported view for complex obs (" + obs.getUuid() + "). Reverting to the default view: '" + ComplexObsHandler.RAW_VIEW + "'".
```

Which test is failing?

Reply:
https://github.com/openmrs/openmrs-module-attachments/blob/7d0466fa3043f25f7bcee7a2d2f879a79573c856/omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java#L86
And the details are in the travis logs.